packages/grid-material/src/GridMaterial.ts

```diff
@@ -0,0 +1,188 @@
+import { BaseMaterial, Engine, MathUtil, Shader } from "oasis-engine";
+
+/**
+ * Grid Material.
+ */
+export class GridMaterial extends BaseMaterial {
+  private static _farClipProperty = Shader.getPropertyByName("u_far");
+  private static _nearClipProperty = Shader.getPropertyByName("u_near");
+  private static _primaryScaleProperty = Shader.getPropertyByName("u_primaryScale");
+  private static _secondaryScaleProperty = Shader.getPropertyByName("u_secondaryScale");
+  private static _gridIntensityProperty = Shader.getPropertyByName("u_gridIntensity");
+  private static _axisIntensityProperty = Shader.getPropertyByName("u_axisIntensity");
+  private static _flipProgressProperty = Shader.getPropertyByName("u_flipProgress");
+
+  /**
+   * Near clip plane - the closest point to the camera when rendering occurs.
+   */
+  get nearClipPlane(): number {
+    return this.shaderData.getFloat(GridMaterial._nearClipProperty);
+  }
+
+  set nearClipPlane(value: number) {
+    this.shaderData.setFloat(GridMaterial._nearClipProperty, value);
+  }
+
+  /**
+   * Far clip plane - the furthest point to the camera when rendering occurs.
+   */
+  get farClipPlane(): number {
+    return this.shaderData.getFloat(GridMaterial._farClipProperty);
+  }
+
+  set farClipPlane(value: number) {
+    this.shaderData.setFloat(GridMaterial._farClipProperty, value);
+  }
+
+  /**
+   * Primary Scale
+   */
+  get primaryScale(): number {
+    return this.shaderData.getFloat(GridMaterial._primaryScaleProperty);
+  }
+
+  set primaryScale(value: number) {
+    this.shaderData.setFloat(GridMaterial._primaryScaleProperty, value);
+  }
+
+  /**
+   * Secondary Scale
+   */
+  get secondaryScale(): number {
+    return this.shaderData.getFloat(GridMaterial._secondaryScaleProperty);
+  }
+
+  set secondaryScale(value: number) {
+    this.shaderData.setFloat(GridMaterial._secondaryScaleProperty, value);
+  }
```

Comment:
`primaryScale` and `secondaryScale` is hard to understand for developer, can you explain further？